highlight/HighlightsFromPointsExplainer.md

```diff
@@ -11,61 +11,494 @@ This document is intended as a starting point for engaging the community and sta
+    }
+}
+
+function removeActiveHighlight(highlight) {
+    let username = highlightToUsername.get(highlight);
+    if (username != undefined) {
+        let activeHighlightName = 'active-selection-highlight-' + username;
+        CSS.highlights.delete(activeHighlightName);
+        currentActiveHighlightUsernames.delete(username);
+        if (currentActiveHighlightUsernames.size == 0) {
+            cursorBox.style.display = 'none';
+        }
+        else {
+            cursorBox.textContent = buildActiveUsernamesLabel();
+        }
+    }
+}
+
+for (highlight of highlightToUsername.keys()) {
+    highlight.addEventListener('mouseover', (event) => {
+        createActiveHighlight(event.currentTarget, event.pageX, event.pageY);
+    });
+
+    highlight.addEventListener('mouseout', (event) => {
+        removeActiveHighlight(event.currentTarget);        
+    });
+}
+```
+
+In contrast to the solutions presented before, in `createActiveHighlights` now we don't have to loop over the highlights affected, but we have to do it beforehand to set all the necessary event listeners.
+Also notice how now we have to be careful and keep track of the active highlights since we're not rebuilding them in the same step, introducing some extra code on that front that wasn't required before.
+
+However, each of these event-based options carry significant complexity from a specification point of view. Option 3 mentioned above would be the most suitable one, but the event phases are hard to define. Let's suppose that the user moves the mouse over the Highlight `h1` set on element `e1`, and an event listener was set on `h1`. The phases should look as follows:
+
+    Capture phase e1 pointermove - target:e1 currentTarget:e1
+    Target phase h1 pointermove - target:e1 currentTarget:h1
+    Bubbling phase e1 pointermove - target:e1 currentTarget:e1
+
+A workaround is needed here where `event.target` should stay as the element `e1` because pointer events can only target elements, but actually the target was the highlight, which doesn't really follow the definition for `event.target`.
+Furthermore, when `currentTarget == target`, it should be target phase, this would imply the event phases work in a new way and it could break current site implementations or introduce inconsistencies with current specifications.
+
+In conclusion, the `highlightsFromPoint()` API is much simpler to implement and specify, it doesn't need these kind of workarounds nor creates these open questions, and still satisfies the use cases adequately.
+
+### Promise-based API
+
+Another asynchronous variation for `highlightsFromPoint` could be returning a [Promise](https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-promise-objects), which returns the highlights present at the given point upon resolving.
+This is some example code for how to implement the online collaboration website with it:
+
+```html
+function createActiveHighlights(x, y) {

```

Comment:
`createActiveHighlights` would probably return `highlightsPromise` so that callers could do await/.then() with the result.